simpleviewinc/tap-events-force
# feat: Session List View

**Tickets**
* [ZEN-498](https://jira.simpleviewtools.com/browse/ZEN-498)
* [ZEN-511](https://jira.simpleviewtools.com/browse/ZEN-511)

**Updates / Notes**
* Speaking with Ben, the Desktop view will now display the session day above each section
* This makes it similar to how it looks on mobile, only it does not have the day name


> **IMPORTANT**
> * Please review [this PR](https://github.com/simpleviewinc/tap-events-force/pull/113) first
>   * This PR is branched off of that one
> * This PR should be tested along with [this PR](https://github.com/simpleviewinc/keg-hub/pull/105)
>   * The two PR's should be merge in at the same time 


[Semver-Status] minor

## Context

We need to update the Sessions Day navigation to match the updated mock-ups
See this ticket => https://jira.simpleviewtools.com/browse/ZEN-472
 

## Goal

* Utilize react-native sectionList
* On Desktop Web show gray line for section titles
* On Mobile show Day number for section titles
* Should also include the name of the day
* On day toggle, it should scroll to the section for that day
* On Desktop web, the day toggle day number needs to be updated when the gray line that separates days touches the bottom of the day toggle component

## Update 1/20/2021
- removed localstorage plugin for dayNumber
- updated sessionList to use setDay instead of decrement/increment
- updated `rollup.config.js`
   * the `kegComponents` path was outdated

## Updates
* Added padding to the bottom of the sessions list to allow for scrolling to the last session day
  * I could not get it to be perfect, and not allow scrolling past the last row
    * This is because the rows have a dynamic height, and can change based on the screen width
    * From my testing, in most cases you can not completely scroll all items out of the screen
      * Typically there's at least some part of the items displayed
    * I messed with this for hours, not much more I can do beyond what it currently is
* Updated some styles to the empty day text
* Updated to show the `no-sessions-exist` component when filtering
  * This is instead of completely removing days where no sessions exist when filtering
  * Completely removing the day makes the UI inconsistent, and could confuse the user
    * For example if we removed day 2 when filtering, and user did not realize it
    * They may think there is **NO DAY 2**
    * So instead it just works like normal when there are no sessions for a day
      * It just displays the `no-sessions-exist` component
  * Cleaned up lots of styles
* On Mobile 
  * Hides the navigation arrows
  * Added the Day name to the header
* Also see changes in this [this PR](https://github.com/simpleviewinc/keg-hub/pull/105)

## Testing (1/21/2021)
* run `keg consumer pack run docker.pkg.github.com/simpleviewinc/keg-packages/keg-test-consumer:develop`
* navigate to http://consumer-develop.local.kegdev.xyz/

**Test 1:** update dayNum when content changes
    * `NOTE: different browsers behaves differently in this scenario but the outcome for us should still be the same` - I'm using `Chrome` for the purpose of this test
* scroll down to day 3
   * [image](https://user-images.githubusercontent.com/3317835/105414819-3b017b00-5bf5-11eb-8ada-e3422b57d9ac.png)
* open up the filter modal and filter by `analytics` label
   * [image](https://user-images.githubusercontent.com/3317835/105414874-5076a500-5bf5-11eb-81db-cb3401fa2de7.png)
   * verify that the `dayNum` header is on the correct day based on your current position
       * [image](https://user-images.githubusercontent.com/3317835/105415207-b2cfa580-5bf5-11eb-8337-d745485d46ea.png)
  * hit `Clear` - to clear the filters
  * verify that the `dayNum` now says `Day 2`
      * [image](https://user-images.githubusercontent.com/3317835/105415315-ced34700-5bf5-11eb-9f1b-3f3cfdd30bf6.png)



## Testing


**Setup**
  * Pull this PR => `keg evf && keg pr 114`   

**Test 1**
* ~~Run the Tap's docker package => `keg evf pack run docker.pkg.github.com/simpleviewinc/keg-packages/evf:zen-498-session-list-view`~~
  * ~~Navigate your browser to http://evf-zen-498-session-list-view.local.kegdev.xyz/~~
* Use this image instead => `keg consumer pack run docker.pkg.github.com/simpleviewinc/keg-packages/keg-test-consumer:develop` 
  * Scroll down in the Session list a bit
    * Ensure the Session list header becomes sticky, and follows you as you scroll
  * Try to scroll from one day to the next
    * Ensure the Sticky header updates as you scroll
  * Try using the arrows to toggle between days
    * Ensure the app scrolls to the correct day and section
  * Try doing both toggling and scrolling in different intervals
    * Ensure everything continues to work as expected
  * Ensure the Sessions list displays the same as in the mockups
    * [Mobile](https://www.figma.com/proto/fgHHqc0OSMlnarOotmgsR5/Sessions_mobile-scroll-v1?node-id=409%3A321&scaling=scale-down)
    * [Desktop](https://www.figma.com/proto/WUCyQjgM0ogsQNmwkaFECF/Sessions_desktop-scroll-v3?node-id=1425%3A2020&scaling=min-zoom)
  * Ensure no errors are throw, and everything works as expected
